src/main/java/com/diffplug/gradle/oomph/WorkspaceRegistry.java

```diff
@@ -99,6 +100,13 @@ public File workspaceDir(Project project, File ideDir) {
 		return workspaceDir(project.getRootProject().getName(), ideDir);
 	}
 
+	/** Returns the workspace directory appropriate for the  */
+	public File workspaceDir(Project project, Task task) {
+		File rootDir = project.getRootDir();
+		String taskName = task.getName();
+		return workspaceDir(taskName, rootDir);
+	}
+
```

Comment:
I think this should be either:

```java
name = task.getName()
dir = task.getProject.getDir()
    OR
name = task.getPath()
dir = task.getProject().getRootDir()
```

There can be more than one task with the same name.  Either of the above is unambiguous,  but the current implementation is a little ambiguous.  Also, since task has `getProject`, I wouldn't pass the project as an arg.